test/scip/testdata/field_inheritance.snapshot.rb

```diff
@@ -0,0 +1,197 @@
+ # typed: true
+ 
+ # First, check that instance variables are propagated through
+ # the inheritance chain.
+ 
+ class C1
+#      ^^ definition [..] C1#
+   def set_ivar
+#      ^^^^^^^^ definition [..] C1#set_ivar().
+     @f = 1
+#    ^^ definition [..] C1#`@f`.
+     return
+   end
+ end
+ 
+ class C2 < C1
+#      ^^ definition [..] C2#
+#           ^^ definition [..] C1#
+   def get_inherited_ivar
+#      ^^^^^^^^^^^^^^^^^^ definition [..] C2#get_inherited_ivar().
+     return @f
+#    ^^^^^^^^^ reference [..] C1#`@f`.
+   end
+ 
+   def set_inherited_ivar
+#      ^^^^^^^^^^^^^^^^^^ definition [..] C2#set_inherited_ivar().
+     @f = 10
+#    ^^ definition [..] C1#`@f`.
+     return
+   end
+ 
+   def set_new_ivar
+#      ^^^^^^^^^^^^ definition [..] C2#set_new_ivar().
+     @g = 1
+#    ^^ definition [..] C2#`@g`.
+     return
+   end
+ 
+   def get_new_ivar
+#      ^^^^^^^^^^^^ definition [..] C2#get_new_ivar().
+     return @g
+#    ^^^^^^^^^ reference [..] C2#`@g`.
+   end
+ end
+ 
+ class C3 < C2
+#      ^^ definition [..] C3#
+#           ^^ definition [..] C2#
+   def refs
+#      ^^^^ definition [..] C3#refs().
+     @f = @g
+#    ^^ definition [..] C1#`@f`.
+#         ^^ reference [..] C2#`@g`.
+     return
+   end
+ end
+ 
+ def c_check
+#    ^^^^^^^ definition [..] Object#c_check().
+   C1.new.instance_variable_get(:@f)
+#  ^^ reference [..] C1#
+#     ^^^ reference [..] Class#new().
+#         ^^^^^^^^^^^^^^^^^^^^^ reference [..] Kernel#instance_variable_get().
+   C2.new.instance_variable_get(:@f)
+#  ^^ reference [..] C2#
+#     ^^^ reference [..] Class#new().
+#         ^^^^^^^^^^^^^^^^^^^^^ reference [..] Kernel#instance_variable_get().
+   C2.new.instance_variable_get(:@g)
+#  ^^ reference [..] C2#
+#     ^^^ reference [..] Class#new().
+#         ^^^^^^^^^^^^^^^^^^^^^ reference [..] Kernel#instance_variable_get().
+   C3.new.instance_variable_get(:@f)
+#  ^^ reference [..] C3#
+#     ^^^ reference [..] Class#new().
+#         ^^^^^^^^^^^^^^^^^^^^^ reference [..] Kernel#instance_variable_get().
+   C3.new.instance_variable_get(:@g)
+#  ^^ reference [..] C3#
+#     ^^^ reference [..] Class#new().
+#         ^^^^^^^^^^^^^^^^^^^^^ reference [..] Kernel#instance_variable_get().
+   return
+ end
+ 
+ # Now, check that class variables work as expected.
+ 
+ class D1
+#      ^^ definition [..] D1#
+   @@d1_v = 0
+#  ^^^^^^ definition [..] `<Class:Object>`#`@@d1_v`.
```

Comment:
TODO: Fix this. It should be `<Class:D>#@@d1_v`. It seems to work fine with smaller examples, so unsure what is going on here.